### PR TITLE
Fix posix logic that was causing panics

### LINKED ISSUE #445: Panic at src/builtins/core/timezone.rs:399:21: attempt to subtract with overflow
Triggered with the following code:

```rust
var dstStart = new Temporal.PlainDateTime(2020, 3, 8, 2, 30);
dstStart.toZonedDateTime("America/Los_Angeles", {disambiguation: "compatible"})
```

Full backtrace:

```
thread '<unnamed>' panicked at ../../third_party/rust/chromium_crates_io/vendor/temporal_rs-v0_0_11/src/builtins/core/timezone.rs:399:21:
attempt to subtract with overflow
stack backtrace:

   0: __rustc::rust_begin_unwind
             at ./../../third_party/rust-toolchain/lib/rustlib/src/rust/library/std/src/panicking.rs:697:5
   1: core::panicking::panic_fmt
             at ./../../third_party/rust-toolchain/lib/rustlib/src/rust/library/core/src/panicking.rs:75:14
   2: core::panicking::panic_const::panic_const_sub_overflow
             at ./../../third_party/rust-toolchain/lib/rustlib/src/rust/library/core/src/panicking.rs:175:17
   3: temporal_rs::builtins::core::timezone::TimeZone::disambiguate_possible_epoch_nanos
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_rs-v0_0_11/src/builtins/core/timezone.rs:399:21
   4: temporal_rs::builtins::core::timezone::TimeZone::get_epoch_nanoseconds_for
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_rs-v0_0_11/src/builtins/core/timezone.rs:210:14
   5: temporal_rs::builtins::core::datetime::PlainDateTime::to_zoned_date_time_with_provider
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_rs-v0_0_11/src/builtins/core/datetime.rs:886:34
   6: temporal_rs::builtins::compiled::plain_date_time::<impl temporal_rs::builtins::core::datetime::PlainDateTime>::to_zoned_date_time
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_rs-v0_0_11/src/builtins/compiled/plain_date_time.rs:17:14
   7: temporal_capi::plain_date_time::ffi::PlainDateTime::to_zoned_date_time
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_capi-v0_0_11/src/plain_date_time.rs:341:18
   8: temporal_rs_PlainDateTime_to_zoned_date_time
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_capi-v0_0_11/src/plain_date_time.rs:3:1
   9: _ZN2v88internal23JSTemporalPlainDateTime15ToZonedDateTimeEPNS0_7IsolateENS0_12DirectHandleIS1_EENS4_INS0_6ObjectEEES7_
             at ./../../third_party/rust/chromium_crates_io/vendor/temporal_capi-v0_0_11/bindings/cpp/temporal_rs/PlainDateTime.hpp:424:17
  10: _ZN2v88internalL58Builtin_Impl_TemporalPlainDateTimePrototypeToZonedDateTimeENS0_16BuiltinArgumentsEPNS0_7IsolateE
             at ./../../src/builtins/builtins-temporal.cc:364:1
  11: _ZN2v88internal53Builtin_TemporalPlainDateTimePrototypeToZonedDateTimeEiPmPNS0_7IsolateE
             at ./../../src/builtins/builtins-temporal.cc:364:1
  12: Builtins_CEntry_Return1_ArgvOnStack_BuiltinExit
```

### PR DESCRIPTION
**Pull request by nekevss**
Closes #445 and #199.

We were not properly resolving the offset for the POSIX time zone for epoch seconds.

This updates the logic to correctly handling determining which offset we are in.

### CODE REVIEW

File: src/builtins/compiled/plain_date_time.rs [hunk 1]
```diff
@@ -17,3 +17,24 @@ impl PlainDateTime {
         self.to_zoned_date_time_with_provider(time_zone, disambiguation, &*TZ_PROVIDER)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "tzdb")]
+    #[test]
+    fn to_zoned_date_time_edge_cases() {
```
Comment (Manishearth) [inline src/builtins/compiled/plain_date_time.rs:25]: nit: some comments explaining what the actual details of the edge case are

Comment (nekevss) [reply src/builtins/compiled/plain_date_time.rs:25]: I added a note to the test and also added a couple more comments to the code to hopefully help with reasoning about the code.

File: src/tzdb.rs [hunk 2]
```diff
@@ -344,29 +346,80 @@ fn resolve_posix_tz_string_for_epoch_seconds(
...
+        calculate_transition_seconds_for_year(year, dst_variant.start_date, std_offset);
+    let dst_end_seconds =
+        calculate_transition_seconds_for_year(year, dst_variant.end_date, dst_offset);
+
+    // Need to determine if the range being tested is standard or savings time.
+    let dst_is_inversed = dst_end_seconds < dst_start_seconds;
+
+    // We have potentially to different variations of the DST start and end time.
+    //
+    // Northern hemisphere: dst_start -> dst_end
```
Comment (Manishearth) [inline src/tzdb.rs:363]: I assume this also handles Ireland's backwards DST

**Review by Manishearth — APPROVED** [review]

File: src/tzdb.rs (hunk 2 above)
Comment (nekevss) [reply src/tzdb.rs:363]: Short answer: I'm not sure

Longer answer:

That's the one caveat to this current approach that will need to be checked.

Currently, it's not too much of an issue though because most of our source data is all in rearguard. So I didn't want to add handling for the vanguard format without a good way to test it for now.
